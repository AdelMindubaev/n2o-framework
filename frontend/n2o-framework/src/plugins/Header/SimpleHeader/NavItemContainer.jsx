import React from 'react';
import PropTypes from 'prop-types';
import { Link, NavLink } from 'react-router-dom';
import cx from 'classnames';
import { isEmpty, get } from 'lodash';

import {
  NavItem,
  UncontrolledDropdown,
  DropdownToggle,
  DropdownMenu,
  DropdownItem,
} from 'reactstrap';
import SecurityCheck from '../../../core/auth/SecurityCheck';

/**
 * Контейнер navItem'ов, в зависимости от type, создает внутри линк, дропдаун или текст
 * @param {object} props - пропсы
 * @param {object} props.item  - объект, пропсы которого перейдут в item. Например, для ссыллок {id, label, href,type, link, linkType}
 * @param {string} props.activeId  - id активного item'a
 * @param {string} props.  - id активного item'a
 */
const NavItemContainer = ({
  item,
  activeId,
  type,
  sidebarOpen,
  options,
  direction,
}) => {
  const getInnerLink = (item, className) => (
    <NavLink exact className="nav-link" to={item.href} activeClassName="active">
      {item.icon && <NavItemIcon />}
      {item.label}
    </NavLink>
  );

  const NavItemIcon = () => <i className={cx('mr-1', item.icon)} />;

  const handleLink = (item, className) => {
    if (item.linkType === 'outer') {
      return (
        <NavItem>
          {item.icon && <NavItemIcon />}
          <a className={cx('nav-link', className)} href={item.href}>
            {item.label}
          </a>
        </NavItem>
      );
    } else {
      return (
        <NavItem>
          <NavLink
            exact
            className={cx('nav-link', className)}
            to={item.href}
            activeClassName="active"
          >
            {item.icon && <NavItemIcon />}
            {item.label}
          </NavLink>
        </NavItem>
      );
    }
  };

  const handleLinkDropdown = (item, dropdownItems) => {
    return (
      <UncontrolledDropdown nav inNavbar direction={direction}>
        <DropdownToggle nav caret>
          {item.icon && <NavItemIcon />}
          {item.label}
        </DropdownToggle>
        <DropdownMenu right={get(options, 'right', false)}>
          {dropdownItems}
        </DropdownMenu>
      </UncontrolledDropdown>
    );
  };

  let dropdownItems = [];
  if (item.type === 'dropdown' && !sidebarOpen) {
    dropdownItems = item.subItems.map((child, i) => (
      <DropdownItem>{handleLink(child, 'dropdown-item')}</DropdownItem>
    ));
    if (
      item.type === 'dropdown' &&
      item.subItems.length > 1 &&
      type === 'sidebar'
    ) {
      dropdownItems = [
        <DropdownItem key={-1} onClick={e => e.preventDefault()}>
          {item.icon && <NavItemIcon />}
          <a className="dropdown-item">{item.oldLabel || item.label}</a>
        </DropdownItem>,
        ...dropdownItems,
      ];
    }
  } else if (type === 'sidebar' && item.type === 'dropdown' && sidebarOpen) {
    const defaultLink = item => (
      <Link className="dropdown-item" to={item.href}>
        {item.icon && <NavItemIcon />}
        {item.label}
      </Link>
    );
    const linkItem = item =>
      item.linkType === 'outer'
        ? defaultLink(item)
        : getInnerLink(item, 'dropdown-item');
    dropdownItems = item.subItems.map((item, i) => (
      <DropdownItem> {linkItem(item)} </DropdownItem>
    ));
  }

  return (
    (item.type === 'dropdown' &&
      !sidebarOpen &&
      handleLinkDropdown(item, dropdownItems)) ||
    (item.type === 'link' && handleLink(item)) ||
    (item.type === 'text' && (
      <NavItem>
        {item.icon && <NavItemIcon />}
        <span className="nav-link">{item.label}</span>
      </NavItem>
    )) ||
    null
  );
};

NavItemContainer.propTypes = {
  item: PropTypes.shape({
    label: PropTypes.string,
    href: PropTypes.string,
    icon: PropTypes.string,
    linkType: PropTypes.oneOf(['inner', 'outer']),
    withSubMenu: PropTypes.bool,
    subItems: PropTypes.array,
  }),
  type: PropTypes.oneOf(['header', 'sidebar']),
  open: PropTypes.bool,
  direction: PropTypes.string,
};

NavItemContainer.defaultProps = {
  direction: 'bottom',
};

export default NavItemContainer;
