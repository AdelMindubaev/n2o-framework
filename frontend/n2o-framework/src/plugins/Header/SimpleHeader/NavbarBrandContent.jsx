import React from 'react';
import PropTypes from 'prop-types';
/**
 * Рендер бренда и лого
 * @param props - пропсы
 * @param {string | element} props.brand  - брэнд(текст)
 * @param {string | element} props.brandImage - брэнд(изображение)
 */
const NavbarBrandContent = ({ brand, brandImage }) => {
  const img =
    brandImage && typeof brandImage === 'string' ? (
      <img
        src={brandImage}
        className="n2o-brand__image d-inline-block align-top"
      />
    ) : (
      brandImage
    );

  return (
    <React.Fragment>
      {img}
      {brand}
    </React.Fragment>
  );
};

NavbarBrandContent.propTypes = {
  brand: PropTypes.oneOfType([PropTypes.string, PropTypes.element]),
  brandImage: PropTypes.oneOfType([PropTypes.string, PropTypes.element]),
};

export default NavbarBrandContent;
