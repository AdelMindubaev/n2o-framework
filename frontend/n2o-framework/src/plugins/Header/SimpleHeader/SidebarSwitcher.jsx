import React from 'react'
import PropTypes from 'prop-types'
import classNames from 'classnames'

export function SidebarSwitcher({
    defaultIcon,
    toggledIcon,
    sidebarOpen,
    toggleSidebar,
}) {
    return (
        <i
            className={classNames('n2o-sidebar-switcher', sidebarOpen ? toggledIcon : defaultIcon)}
            aria-hidden="true"
            onClick={toggleSidebar}
        />
    )
}
SidebarSwitcher.propTypes = {
    defaultIcon: PropTypes.string,
    toggledIcon: PropTypes.string,
    sidebarOpen: PropTypes.bool,
    toggleSidebar: PropTypes.func,
}
