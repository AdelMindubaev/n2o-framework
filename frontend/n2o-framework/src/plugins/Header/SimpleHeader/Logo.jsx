import React from 'react'
import PropTypes from 'prop-types'
import classNames from 'classnames'
import NavbarBrand from 'reactstrap/lib/NavbarBrand'

import { NavbarBrandContent } from './NavbarBrandContent'

export function Logo({ title, className, style, href, src, showContent, isMiniView }) {
    return (
        <section
            className={classNames(
                'n2o-header-logo d-flex flex-wrap justify-content-center',
                className,
                {
                    visible: showContent,
                    mini: isMiniView,
                },
            )}
            style={style}
        >
            {src && (
                <NavbarBrand className="n2o-brand" href={href}>
                    <NavbarBrandContent brandImage={src} />
                </NavbarBrand>
            )}
            {title && (
                <a href={href} className="navbar-brand">
                    {title}
                </a>
            )}
        </section>
    )
}

Logo.propTypes = {
    title: PropTypes.string,
    className: PropTypes.string,
    style: PropTypes.object,
    href: PropTypes.string,
    src: PropTypes.string,
    showContent: PropTypes.bool,
    isMiniView: PropTypes.bool,
}

Logo.defaultProps = {
    href: '/',
    showContent: true,
}
