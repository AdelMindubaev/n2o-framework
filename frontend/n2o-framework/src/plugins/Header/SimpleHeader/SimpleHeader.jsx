import React from 'react';

import isEmpty from 'lodash/isEmpty';

import PropTypes from 'prop-types';
import Navbar from 'reactstrap/lib/Navbar';
import Nav from 'reactstrap/lib/Nav';
import NavItem from 'reactstrap/lib/NavItem';
import InputGroup from 'reactstrap/lib/InputGroup';
import InputGroupAddon from 'reactstrap/lib/InputGroupAddon';
import Input from 'reactstrap/lib/Input';
import map from 'lodash/map';
import NavbarBrand from 'reactstrap/lib/NavbarBrand';
import NavbarToggler from 'reactstrap/lib/NavbarToggler';
import Collapse from 'reactstrap/lib/Collapse';

import NavbarBrandContent from './NavbarBrandContent';
import NavItemContainer from './NavItemContainer';
import SearchBarContainer from '../../../components/snippets/SearchBar/SearchBarContainer';

/**
 * Хедер-плагин
 * @param {Object} props - пропсы
 * @param {string|element} props.brand - брэнд
 * @param {string|element} props.brandImage - изображение брэнда
 * @param {array} props.items - элементы навбар-меню (левое меню)
 * @param {boolean} props.fixed - фиксированный хэдер или нет
 * @param {array} props.extraItems - элементы навбар-меню (правое меню)
 * @param {boolean} props.collapsed - находится в состоянии коллапса или нет
 * @param {boolean} props.search - есть поле поиска / нет поля поиска
 * @param {boolean} props.color - стиль хэдера (default или inverse)
 * @param {boolean} props.className - css-класс
 * @param {boolean} props.style - объект стиля
 * @example
 * //каждый item состоит из id {string}, label {string}, type {string} ('text', 'type' или 'dropdown'),
 * //href {string}(для ссылок), linkType {string}(для ссылок; значения - 'outer' или 'inner')
 * //badge {string} (текст баджа), badgeColor {string} (цвет баджа), target {string} ('_blank' или null)
 * //subItems {array} (массив из элементов дропдауна)
 *<SimpleHeader  items = { [
 *     {
 *       id: 'link',
 *       label: 'link',
 *       href: '/test',
 *       type: 'link',
 *       target: '_blank',
 *     },
 *     {
 *       id: 'dropdown',
 *       label: 'dropdown',
 *       type: 'dropdown',
 *       subItems: [{id: 'test1',label: 'test1', href: '/', badge: 'badge1', badgeColor: 'color1'},
 *       {id: 'test123', label: 'test1', href: '/',  badge: 'badge2', badgeColor: 'color2'}]
 *     },
 *     {
 *       id: 'test',
 *       label: 'test',
 *       type: 'dropdown',
 *       subItems: [{id: 'test123s',label: 'test1', href: '/', badge: 'badge1', badgeColor: 'color1'},
 *       {id: 'test12asd3',label: 'test1', href: '/',  badge: 'badge2', badgeColor: 'color2'}]
 *     }
 *     ] }
 *     extraItems = { [
 *     {
 *       id: "213",
 *       label: 'ГКБ №7',
 *       type: 'text',
 *     },
 *     {
 *       id: "2131",
 *       label: 'Постовая медсестра',
 *       type: 'dropdown',
 *       subItems: [{label: 'test1', href: '/', linkType: 'inner'}, {label: 'test1', href: '/'}]
 *     },
 *     {
 *       id: "2131",
 *       label: 'admin',
 *       type: 'dropdown',
 *       subItems: [{label: 'test1', href: '/'}, {label: 'test1', href: '/'}]
 *     }
 *     ] }
 *    brand="N2O"
 *    brandImage= "http://getbootstrap.com/assets/brand/bootstrap-solid.svg"
 *    activeId={"test123"}/>
 *
 */

class SimpleHeader extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      isOpen: false,
    };

    this.toggle = this.toggle.bind(this);
  }

  toggle() {
    this.setState({
      isOpen: !this.state.isOpen,
    });
  }

  render() {
    const {
      color,
      fixed,
      items,
      activeId,
      extraItems,
      brandImage,
      brand,
      style,
      className,
      search,
      homePageUrl,
    } = this.props;

    const isInversed = color === 'inverse';
    const navColor = isInversed ? 'primary' : 'light';
    const mapItems = (items, options) =>
      map(items, (item, i) => (
        <NavItemContainer
          key={i}
          item={item}
          activeId={activeId}
          options={options}
        />
      ));

    const navItems = mapItems(items);
    const extraNavItems = mapItems(extraItems, { right: true });

    return (
      <div
        style={style}
        className={`navbar-container-${
          fixed ? 'fixed' : 'relative'
        } ${className} n2o-header n2o-header-${color} `}
      >
        <Navbar
          color={navColor}
          light={!isInversed}
          dark={isInversed}
          expand="md"
        >
          {brandImage && (
            <NavbarBrand className="n2o-brand" href={homePageUrl}>
              <NavbarBrandContent brandImage={brandImage} />
            </NavbarBrand>
          )}
          {brand && (
            <a href={homePageUrl} className="navbar-brand">
              {brand}
            </a>
          )}
          {!isEmpty(items) && <NavbarToggler onClick={this.toggle} />}
          <Collapse isOpen={this.state.isOpen} navbar>
            <Nav className="main-nav" navbar>
              {navItems}
            </Nav>
            <Nav className="ml-auto main-nav-extra" navbar>
              {extraNavItems}
              {search && <SearchBarContainer />}
            </Nav>
          </Collapse>
        </Navbar>
      </div>
    );
  }
}

SimpleHeader.propTypes = {
  /**
   * ID активного элемента
   */
  activeId: PropTypes.string,
  /**
   * Бренд хедера
   */
  brand: PropTypes.oneOfType([PropTypes.string, PropTypes.element]),
  /**
   * Картинка бренда
   */
  brandImage: PropTypes.oneOfType([PropTypes.string, PropTypes.element]),
  /**
   * Элементы хедера
   */
  items: PropTypes.arrayOf(
    PropTypes.shape({
      id: PropTypes.string.isRequired,
      label: PropTypes.string.isRequired,
      href: PropTypes.string,
      linkType: PropTypes.oneOf(['inner', 'outer']),
      type: PropTypes.oneOf(['dropdown', 'link', 'text']).isRequired,
      subItems: PropTypes.array,
      badge: PropTypes.string,
      badgeColor: PropTypes.string,
      target: PropTypes.string,
    })
  ),
  /**
   * Extra элементы хедера
   */
  extraItems: PropTypes.arrayOf(
    PropTypes.shape({
      id: PropTypes.string.isRequired,
      label: PropTypes.string.isRequired,
      href: PropTypes.string,
      linkType: PropTypes.oneOf(['inner', 'outer']),
      type: PropTypes.oneOf(['dropdown', 'link', 'text']).isRequired,
      subItems: PropTypes.array,
      badge: 'badge',
      badgeColor: 'color',
      target: PropTypes.string,
    })
  ),
  /**
   * Строка поиска
   */
  search: PropTypes.bool,
  /**
   * Адрес ссылка бренда
   */
  homePageUrl: PropTypes.string,
  /**
   * Цвет хедера
   */
  color: PropTypes.oneOf(['inverse', 'default']),
  /**
   * Флаг фиксированного хедера
   */
  fixed: PropTypes.bool,
  /**
   * Флаг сжатости хедера
   */
  collapsed: PropTypes.bool,
  /**
   * Класс
   */
  className: PropTypes.string,
  /**
   * Стили
   */
  style: PropTypes.object,
};

SimpleHeader.defaultProps = {
  color: 'default',
  fixed: true,
  homePageUrl: '/',
  collapsed: true,
  className: '',
  items: [],
  extraItems: [],
  search: false,
  style: {},
  list: [],
};

export default SimpleHeader;
