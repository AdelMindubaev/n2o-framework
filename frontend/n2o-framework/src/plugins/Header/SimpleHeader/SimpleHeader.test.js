import React from 'react';
import SimpleHeader from './SimpleHeader';
import { mount } from 'enzyme';
import { BrowserRouter as Router } from 'react-router-dom';
import { shallowToJson } from 'enzyme-to-json';
import MetaJSON from './simpleHeaderData';

const props = MetaJSON;

const setup = (propOverrides, defaultProps = props) => {
  const props = Object.assign({}, defaultProps, propOverrides);

  const wrapper = mount(
    <Router>
      <SimpleHeader {...props} />
    </Router>
  );

  return {
    props,
    wrapper,
  };
};

describe('<SimpleHeader />', () => {
  it('проверяет создание компонента', () => {
    const { wrapper } = setup();
    wrapper.update();
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
  it('проверяет props', () => {
    const { wrapper } = setup();
    const expectedValue = props.items[1];
    wrapper.update();
    expect(wrapper.find('NavItemContainer').get(1).props.item).toBe(
      expectedValue
    );
  });
  it('проверяет открытие меню', () => {
    const { wrapper } = setup();
    wrapper.update();
    wrapper
      .find('.dropdown-toggle')
      .first()
      .simulate('click');
    expect(
      wrapper
        .find('.dropdown-menu')
        .first()
        .hasClass('show')
    ).toBeTruthy();
  });
  it('проверяет, что активный элемент устанавливается правильно', () => {
    const { wrapper } = setup();
    wrapper.update();
    wrapper
      .find('button.dropdown-item')
      .first()
      .simulate('click');
    expect(
      wrapper
        .find('a.nav-link.dropdown-item')
        .last()
        .hasClass('active')
    ).toBeTruthy();
  });
  it('проверяет баджи', () => {
    const { wrapper } = setup();
    wrapper.update();
    expect(wrapper.find('Badge').exists()).toBeTruthy();
  });
});
