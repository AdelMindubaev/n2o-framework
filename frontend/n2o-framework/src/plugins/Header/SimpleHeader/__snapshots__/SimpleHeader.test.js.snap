// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SimpleHeader /> проверяет создание компонента 1`] = `
<BrowserRouter>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <SimpleHeader
      activeId="menuItem0"
      brand="Лого"
      brandImage="https://avatars0.githubusercontent.com/u/25926683?s=200&v=4"
      className=""
      collapsed={true}
      color="invert"
      extraItems={
        Array [
          Object {
            "id": "text",
            "label": "текст",
            "type": "text",
          },
          Object {
            "id": "dropdown1",
            "label": "список №2",
            "subItems": Array [
              Object {
                "badge": "бадж 1",
                "badgeColor": "warning",
                "href": "/",
                "label": "элемент списка №1",
                "type": "link",
              },
              Object {
                "badge": "бадж 2",
                "badgeColor": "warning",
                "href": "/",
                "label": "элемент списка №2",
                "type": "link",
              },
            ],
            "type": "dropdown",
          },
        ]
      }
      fixed={false}
      items={
        Array [
          Object {
            "href": "/",
            "id": "menuItem0",
            "label": "ссылка",
            "linkType": "outer",
            "security": Object {
              "roles": Array [
                "admin",
              ],
            },
            "type": "link",
          },
          Object {
            "href": "/pageRoute",
            "id": "menuItem1",
            "label": "список",
            "linkType": "inner",
            "subItems": Array [
              Object {
                "badge": "бадж 1",
                "badgeColor": "danger",
                "href": "/pageRoute",
                "id": "menuItem2",
                "label": "Название страницы",
                "linkType": "inner",
                "type": "link",
              },
              Object {
                "badge": "бадж 2",
                "badgeColor": "danger",
                "href": "/pageRoute1",
                "id": "menuItem3",
                "label": "элемент списка №2",
                "linkType": "inner",
                "type": "link",
              },
            ],
            "type": "dropdown",
          },
        ]
      }
      search={false}
      style={Object {}}
    >
      <div
        className="navbar-container-relative  n2o-header n2o-header-invert "
        style={Object {}}
      >
        <Navbar
          color="light"
          dark={false}
          expand="md"
          light={true}
          tag="nav"
        >
          <nav
            className="navbar navbar-expand-md navbar-light bg-light"
          >
            <NavbarBrand
              href="/"
              tag="a"
            >
              <a
                className="navbar-brand"
                href="/"
              >
                <NavbarBrandContent
                  brand="Лого"
                  brandImage="https://avatars0.githubusercontent.com/u/25926683?s=200&v=4"
                >
                  <img
                    className="d-inline-block align-top"
                    src="https://avatars0.githubusercontent.com/u/25926683?s=200&v=4"
                    style={
                      Object {
                        "alignSelf": "center",
                        "display": "inline-block",
                        "margin": "auto",
                        "marginRight": "5px",
                        "maxHeight": 40,
                        "maxWidth": 40,
                      }
                    }
                  />
                  Лого
                </NavbarBrandContent>
              </a>
            </NavbarBrand>
            <NavbarToggler
              onClick={[Function]}
              tag="button"
              type="button"
            >
              <button
                aria-label="Toggle navigation"
                className="navbar-toggler"
                onClick={[Function]}
                type="button"
              >
                <span
                  className="navbar-toggler-icon"
                />
              </button>
            </NavbarToggler>
            <Collapse
              appear={false}
              enter={true}
              exit={true}
              in={false}
              isOpen={false}
              mountOnEnter={false}
              navbar={true}
              onEnter={[Function]}
              onEntered={[Function]}
              onEntering={[Function]}
              onExit={[Function]}
              onExited={[Function]}
              onExiting={[Function]}
              tag="div"
              timeout={350}
              unmountOnExit={false}
            >
              <Transition
                appear={false}
                enter={true}
                exit={true}
                in={false}
                mountOnEnter={false}
                onEnter={[Function]}
                onEntered={[Function]}
                onEntering={[Function]}
                onExit={[Function]}
                onExited={[Function]}
                onExiting={[Function]}
                timeout={350}
                unmountOnExit={false}
              >
                <div
                  aria-expanded="false"
                  className="collapse navbar-collapse"
                  style={Object {}}
                >
                  <Nav
                    navbar={true}
                    tag="ul"
                    vertical={false}
                  >
                    <ul
                      className="navbar-nav"
                    >
                      <NavItemContainer
                        activeId="menuItem0"
                        direction="bottom"
                        item={
                          Object {
                            "href": "/",
                            "id": "menuItem0",
                            "label": "ссылка",
                            "linkType": "outer",
                            "security": Object {
                              "roles": Array [
                                "admin",
                              ],
                            },
                            "type": "link",
                          }
                        }
                        key="0"
                      >
                        <NavItem
                          tag="li"
                        >
                          <li
                            className="nav-item"
                          >
                            <a
                              className="nav-link"
                              href="/"
                            >
                              ссылка
                            </a>
                            <Badge
                              color="secondary"
                              pill={false}
                              tag="span"
                            >
                              <span
                                className="badge badge-secondary"
                              />
                            </Badge>
                          </li>
                        </NavItem>
                      </NavItemContainer>
                      <NavItemContainer
                        activeId="menuItem0"
                        direction="bottom"
                        item={
                          Object {
                            "href": "/pageRoute",
                            "id": "menuItem1",
                            "label": "список",
                            "linkType": "inner",
                            "subItems": Array [
                              Object {
                                "badge": "бадж 1",
                                "badgeColor": "danger",
                                "href": "/pageRoute",
                                "id": "menuItem2",
                                "label": "Название страницы",
                                "linkType": "inner",
                                "type": "link",
                              },
                              Object {
                                "badge": "бадж 2",
                                "badgeColor": "danger",
                                "href": "/pageRoute1",
                                "id": "menuItem3",
                                "label": "элемент списка №2",
                                "linkType": "inner",
                                "type": "link",
                              },
                            ],
                            "type": "dropdown",
                          }
                        }
                        key="1"
                      >
                        <UncontrolledDropdown
                          direction="bottom"
                          inNavbar={true}
                          nav={true}
                        >
                          <Dropdown
                            a11y={true}
                            active={false}
                            addonType={false}
                            direction="bottom"
                            inNavbar={true}
                            isOpen={false}
                            nav={true}
                            setActiveFromChild={false}
                            toggle={[Function]}
                          >
                            <Manager>
                              <li
                                className="dropbottom dropdown nav-item"
                                onKeyDown={[Function]}
                              >
                                <DropdownToggle
                                  aria-haspopup={true}
                                  caret={true}
                                  color="secondary"
                                  nav={true}
                                >
                                  <a
                                    aria-expanded={false}
                                    aria-haspopup={true}
                                    className="dropdown-toggle nav-link"
                                    href="#"
                                    onClick={[Function]}
                                  >
                                    список
                                  </a>
                                </DropdownToggle>
                                <DropdownMenu
                                  flip={true}
                                  right={false}
                                  tag="div"
                                >
                                  <div
                                    aria-hidden={true}
                                    className="dropdown-menu"
                                    role="menu"
                                    tabIndex="-1"
                                  >
                                    <DropdownItem
                                      tag="button"
                                      toggle={true}
                                    >
                                      <button
                                        className="dropdown-item"
                                        onClick={[Function]}
                                        role="menuitem"
                                        tabIndex="0"
                                        type="button"
                                      >
                                        <NavItem
                                          tag="li"
                                        >
                                          <li
                                            className="nav-item"
                                          >
                                            <NavLink
                                              activeClassName="active"
                                              aria-current="page"
                                              className="nav-link dropdown-item"
                                              exact={true}
                                              to="/pageRoute"
                                            >
                                              <Route
                                                exact={true}
                                                path="\\\\/pageRoute"
                                              >
                                                <Link
                                                  aria-current={null}
                                                  className="nav-link dropdown-item"
                                                  replace={false}
                                                  to="/pageRoute"
                                                >
                                                  <a
                                                    aria-current={null}
                                                    className="nav-link dropdown-item"
                                                    href="/pageRoute"
                                                    onClick={[Function]}
                                                  >
                                                    Название страницы
                                                  </a>
                                                </Link>
                                              </Route>
                                            </NavLink>
                                            <Badge
                                              color="danger"
                                              pill={false}
                                              tag="span"
                                            >
                                              <span
                                                className="badge badge-danger"
                                              >
                                                бадж 1
                                              </span>
                                            </Badge>
                                          </li>
                                        </NavItem>
                                      </button>
                                    </DropdownItem>
                                    <DropdownItem
                                      tag="button"
                                      toggle={true}
                                    >
                                      <button
                                        className="dropdown-item"
                                        onClick={[Function]}
                                        role="menuitem"
                                        tabIndex="0"
                                        type="button"
                                      >
                                        <NavItem
                                          tag="li"
                                        >
                                          <li
                                            className="nav-item"
                                          >
                                            <NavLink
                                              activeClassName="active"
                                              aria-current="page"
                                              className="nav-link dropdown-item"
                                              exact={true}
                                              to="/pageRoute1"
                                            >
                                              <Route
                                                exact={true}
                                                path="\\\\/pageRoute1"
                                              >
                                                <Link
                                                  aria-current={null}
                                                  className="nav-link dropdown-item"
                                                  replace={false}
                                                  to="/pageRoute1"
                                                >
                                                  <a
                                                    aria-current={null}
                                                    className="nav-link dropdown-item"
                                                    href="/pageRoute1"
                                                    onClick={[Function]}
                                                  >
                                                    элемент списка №2
                                                  </a>
                                                </Link>
                                              </Route>
                                            </NavLink>
                                            <Badge
                                              color="danger"
                                              pill={false}
                                              tag="span"
                                            >
                                              <span
                                                className="badge badge-danger"
                                              >
                                                бадж 2
                                              </span>
                                            </Badge>
                                          </li>
                                        </NavItem>
                                      </button>
                                    </DropdownItem>
                                  </div>
                                </DropdownMenu>
                              </li>
                            </Manager>
                          </Dropdown>
                        </UncontrolledDropdown>
                      </NavItemContainer>
                    </ul>
                  </Nav>
                  <Nav
                    className="ml-auto"
                    navbar={true}
                    tag="ul"
                    vertical={false}
                  >
                    <ul
                      className="ml-auto navbar-nav"
                    >
                      <NavItemContainer
                        activeId="menuItem0"
                        direction="bottom"
                        item={
                          Object {
                            "id": "text",
                            "label": "текст",
                            "type": "text",
                          }
                        }
                        key="0"
                        options={
                          Object {
                            "right": true,
                          }
                        }
                      >
                        <NavItem
                          tag="li"
                        >
                          <li
                            className="nav-item"
                          >
                            <span
                              className="nav-link"
                            >
                              текст
                            </span>
                          </li>
                        </NavItem>
                      </NavItemContainer>
                      <NavItemContainer
                        activeId="menuItem0"
                        direction="bottom"
                        item={
                          Object {
                            "id": "dropdown1",
                            "label": "список №2",
                            "subItems": Array [
                              Object {
                                "badge": "бадж 1",
                                "badgeColor": "warning",
                                "href": "/",
                                "label": "элемент списка №1",
                                "type": "link",
                              },
                              Object {
                                "badge": "бадж 2",
                                "badgeColor": "warning",
                                "href": "/",
                                "label": "элемент списка №2",
                                "type": "link",
                              },
                            ],
                            "type": "dropdown",
                          }
                        }
                        key="1"
                        options={
                          Object {
                            "right": true,
                          }
                        }
                      >
                        <UncontrolledDropdown
                          direction="bottom"
                          inNavbar={true}
                          nav={true}
                        >
                          <Dropdown
                            a11y={true}
                            active={false}
                            addonType={false}
                            direction="bottom"
                            inNavbar={true}
                            isOpen={false}
                            nav={true}
                            setActiveFromChild={false}
                            toggle={[Function]}
                          >
                            <Manager>
                              <li
                                className="dropbottom dropdown nav-item"
                                onKeyDown={[Function]}
                              >
                                <DropdownToggle
                                  aria-haspopup={true}
                                  caret={true}
                                  color="secondary"
                                  nav={true}
                                >
                                  <a
                                    aria-expanded={false}
                                    aria-haspopup={true}
                                    className="dropdown-toggle nav-link"
                                    href="#"
                                    onClick={[Function]}
                                  >
                                    список №2
                                  </a>
                                </DropdownToggle>
                                <DropdownMenu
                                  flip={true}
                                  right={true}
                                  tag="div"
                                >
                                  <div
                                    aria-hidden={true}
                                    className="dropdown-menu dropdown-menu-right"
                                    role="menu"
                                    tabIndex="-1"
                                  >
                                    <DropdownItem
                                      tag="button"
                                      toggle={true}
                                    >
                                      <button
                                        className="dropdown-item"
                                        onClick={[Function]}
                                        role="menuitem"
                                        tabIndex="0"
                                        type="button"
                                      >
                                        <NavItem
                                          tag="li"
                                        >
                                          <li
                                            className="nav-item"
                                          >
                                            <NavLink
                                              activeClassName="active"
                                              aria-current="page"
                                              className="nav-link dropdown-item"
                                              exact={true}
                                              to="/"
                                            >
                                              <Route
                                                exact={true}
                                                path="\\\\/"
                                              >
                                                <Link
                                                  aria-current="page"
                                                  className="nav-link dropdown-item active"
                                                  replace={false}
                                                  style={Object {}}
                                                  to="/"
                                                >
                                                  <a
                                                    aria-current="page"
                                                    className="nav-link dropdown-item active"
                                                    href="/"
                                                    onClick={[Function]}
                                                    style={Object {}}
                                                  >
                                                    элемент списка №1
                                                  </a>
                                                </Link>
                                              </Route>
                                            </NavLink>
                                            <Badge
                                              color="warning"
                                              pill={false}
                                              tag="span"
                                            >
                                              <span
                                                className="badge badge-warning"
                                              >
                                                бадж 1
                                              </span>
                                            </Badge>
                                          </li>
                                        </NavItem>
                                      </button>
                                    </DropdownItem>
                                    <DropdownItem
                                      tag="button"
                                      toggle={true}
                                    >
                                      <button
                                        className="dropdown-item"
                                        onClick={[Function]}
                                        role="menuitem"
                                        tabIndex="0"
                                        type="button"
                                      >
                                        <NavItem
                                          tag="li"
                                        >
                                          <li
                                            className="nav-item"
                                          >
                                            <NavLink
                                              activeClassName="active"
                                              aria-current="page"
                                              className="nav-link dropdown-item"
                                              exact={true}
                                              to="/"
                                            >
                                              <Route
                                                exact={true}
                                                path="\\\\/"
                                              >
                                                <Link
                                                  aria-current="page"
                                                  className="nav-link dropdown-item active"
                                                  replace={false}
                                                  style={Object {}}
                                                  to="/"
                                                >
                                                  <a
                                                    aria-current="page"
                                                    className="nav-link dropdown-item active"
                                                    href="/"
                                                    onClick={[Function]}
                                                    style={Object {}}
                                                  >
                                                    элемент списка №2
                                                  </a>
                                                </Link>
                                              </Route>
                                            </NavLink>
                                            <Badge
                                              color="warning"
                                              pill={false}
                                              tag="span"
                                            >
                                              <span
                                                className="badge badge-warning"
                                              >
                                                бадж 2
                                              </span>
                                            </Badge>
                                          </li>
                                        </NavItem>
                                      </button>
                                    </DropdownItem>
                                  </div>
                                </DropdownMenu>
                              </li>
                            </Manager>
                          </Dropdown>
                        </UncontrolledDropdown>
                      </NavItemContainer>
                    </ul>
                  </Nav>
                </div>
              </Transition>
            </Collapse>
          </nav>
        </Navbar>
      </div>
    </SimpleHeader>
  </Router>
</BrowserRouter>
`;
